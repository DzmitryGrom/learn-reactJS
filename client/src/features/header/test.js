import React from 'react';
import { mount } from 'enzyme';
import Header from './index';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore(),
  mockStoreInitialized = mockStore({ 
      films: [ 
          {
              "data": [
                {
                  "id": 0,
                  "title": "string",
                  "tagline": "string",
                  "vote_average": 0,
                  "vote_count": 0,
                  "release_date": "string",
                  "poster_path": "string",
                  "overview": "string",
                  "budget": 0,
                  "revenue": 0,
                  "runtime": 0,
                  "genres": [
                    "string"
                  ]
                }
              ],
              "total": 0,
              "offset": 0,
              "limit": 0
            }
      ]
  }); 


describe('tests header', () => {
    it('render  component', () => {
        const wrapper = mount(<Provider store={mockStoreInitialized}><Header/></Provider>),
            component = wrapper.find(Header);
        expect(component).toMatchSnapshot();
        expect(component.find('#movieValue')).toBeTruthy();
        expect(component.find('#movieValue').text()).not.toBe('2 movies found');
        expect(component.find('#movieValue').text()).toBe('1 movies found');
        expect(component.find('.logo')).toBeTruthy();
        expect(component.find('.filter')).toBeTruthy();
     });
})