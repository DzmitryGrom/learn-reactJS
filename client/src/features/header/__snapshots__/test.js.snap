// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests header render  component 1`] = `
<Connect(Header)>
  <Header
    dispatch={[Function]}
    filmsLength={1}
  >
    <div
      className="header"
    >
      <div
        className="header__wrapper"
      >
        <Logo>
          <span
            className="logo"
          >
            netflixroulette
          </span>
        </Logo>
        <Connect(FilterContainer)>
          <FilterContainer
            setSearchByGenre={[Function]}
            setSearchByTitle={[Function]}
            setSearchText={[Function]}
            setSortByRating={[Function]}
            setSortByRelease={[Function]}
          >
            <FilterComponent
              getInputRef={[Function]}
              onButtonClick={[Function]}
              onButtonClickGenre={[Function]}
              onButtonClickRating={[Function]}
              onButtonClickRelease={[Function]}
              onButtonClickTitle={[Function]}
            >
              <div
                className="filter"
              >
                <div
                  className="filter__top"
                >
                  <span
                    className="filter__label"
                  >
                    Find your movie
                  </span>
                  <input
                    className="filter__input"
                    placeholder="Type your value"
                    type="text"
                  />
                  <div
                    className="filter__buttons"
                  >
                    <span
                      className="filter__buttonsLabel"
                    >
                      search by
                    </span>
                    <label
                      htmlFor="btnTitle"
                    >
                      <input
                        id="btnTitle"
                        name="search"
                        onClick={[Function]}
                        type="radio"
                      />
                      <span
                        className="filter__btn"
                      >
                        title
                      </span>
                    </label>
                    <label
                      htmlFor="btnGenre"
                    >
                      <input
                        id="btnGenre"
                        name="search"
                        onClick={[Function]}
                        type="radio"
                      />
                      <span
                        className="filter__btn"
                      >
                        genre
                      </span>
                    </label>
                  </div>
                  <Button
                    modifier="pink"
                    onButtonClick={[Function]}
                    selector="btnSearch"
                    size="big"
                    text="search"
                  >
                    <input
                      className="btn btn_pink btn_big"
                      id="btnSearch"
                      onClick={[Function]}
                      type="button"
                      value="search"
                    />
                  </Button>
                </div>
                <div
                  className="filter__btns"
                >
                  <span
                    className="filter__value"
                    id="btnSortBy"
                  >
                    Sort by
                  </span>
                  <label
                    htmlFor="btnRelease"
                  >
                    <input
                      id="btnRelease"
                      name="sort"
                      onClick={[Function]}
                      type="radio"
                    />
                    <span
                      className="filter__value"
                    >
                      release date
                    </span>
                  </label>
                  <label
                    htmlFor="btnRating"
                  >
                    <input
                      id="btnRating"
                      name="sort"
                      onClick={[Function]}
                      type="radio"
                    />
                    <span
                      className="filter__value"
                    >
                      rating
                    </span>
                  </label>
                </div>
              </div>
            </FilterComponent>
          </FilterContainer>
        </Connect(FilterContainer)>
      </div>
      <div
        className="header__info"
      >
        <div>
          <span
            className="header__value"
            id="movieValue"
          >
            1
             
            movies found
          </span>
        </div>
      </div>
    </div>
  </Header>
</Connect(Header)>
`;
